core/CMakeLists.txt

```diff
@@ -142,3 +142,7 @@ detray_add_library( detray_core core
    "include/detray/utils/type_registry.hpp"
    "include/detray/utils/type_traits.hpp" )
 target_link_libraries( detray_core INTERFACE detray_io vecmem::core detray::Thrust)
+
+# Allow infinite math for LLVM.
+target_compile_options( detray_core
+  INTERFACE $<$<CXX_COMPILER_ID:IntelLLVM>:-fno-finite-math-only> )
```

Comment:
No. I was fine with modifying `CMAKE_CXX_FLAGS` for the build of the tests in this project. But no, `detray::core` should not be forcing compiler specific flags on its clients like this. I really don't like this!

Reply:
> They provide helper functions that allow clients to set up their builds with the flags that the Vc and Thrust developers think are the best for the occasion. 

Do you have a link for this?

Reply:
Let me first point at how we use the interfaces of their helper functions.
  - For Vc we do this:
    * https://github.com/acts-project/algebra-plugins/blob/main/CMakeLists.txt#L134-L141
  - And for Thrust we do the following:
    * https://github.com/acts-project/detray/blob/main/CMakeLists.txt#L157-L162
    * https://github.com/acts-project/traccc/blob/main/CMakeLists.txt#L113-L118

Unfortunately neither of these are perfect, so I myself would prefer a slight modification on what Thrust is doing. Since right now, if we "build" Detray before using it, the Thrust options are frozen for the client.

https://github.com/acts-project/detray/blob/main/cmake/detray-config.cmake.in#L31-L33

And as I realised just now, we'd also need to ensure that traccc and detray use the same flags for Thrust, by adding something like

```cmake
set( DETRAY_THRUST_OPTIONS ${TRACCC_THRUST_OPTIONS}
   CACHE STRING "Thrust setup options to use in the Detray configuration" )
```

to:

https://github.com/acts-project/traccc/blob/main/extern/detray/CMakeLists.txt#L50

(For Thrust this setup, where the configuration gets frozen, is not a huge deal. Since the differences are just for what "backends" should be enabled from TBB, OpenMP, CUDA, etc.)

The implementation of these interfaces is not even all that interesting. Rather, what I would imagine as a functional interface could be a function that would be provided by Detray, and could be used like:

```cmake
detray_set_preferred_flags( detray::core FOO BAR )
```

I.e. the function could receive some flags similar to what `thrust_create_target(...)` is doing, but these flags would not automatically propagate into the exported configuration of a project.